lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1906,6 +1978,12 @@ void glow::optimize(Function *F, CompilationMode mode) {
 
   optimizeReshape(F);
 
+  // Merge Transpose into MatMul.
+  // Run after optimizeReshape to ensure a single Reshapes precedes MatMul.
+  // Run DCE to ensure correct number of node users.
```

Comment:
Why do we need the DCE here?

Reply:
Optimizations do not erase nodes, they just do 'ReplaceAllUsesWith' and relay on DCE to clean up. This corrupts number of users for nodes (after transformation we usually have a newly created optimized use and old 'dead' use), i.e. node->hasOneUse() may return false, while after DCE it'll be true.

Reply:
Oh, wow this is a bug!  Could you please file an issue to track this bug?  Optimizations should not corrupt the graph.

Reply:
Created https://github.com/pytorch/glow/issues/1688